java/sage/MediaServer.java

```diff
@@ -1157,6 +1213,147 @@ else if (tempString.indexOf("XCODE_SETUP ") == 0 && !Sage.EMBEDDED)
+                  tempString.substring(tempString.indexOf(" ") + 1));
+            }
+            else
+            {
+              if (Sage.DBG) System.out.println("MediaServer is writing in remux mode: " +
+                  tempString.substring(tempString.indexOf(" ") + 1));
+
+              StringTokenizer toker = new StringTokenizer(tempString.substring(12), " ");
+
+              String mode = toker.nextToken();
+              int outputFormat = toker.nextToken().equals("TS") ? MPEGParser2.REMUX_TS : MPEGParser2.REMUX_PS;
+
+              if (mode.equals("AUTO"))
+              {
+                boolean isTV = toker.nextToken().equals("TRUE");
+                remuxer = new MediaServerRemuxer(fileChannel, outputFormat, isTV, this);
+              }
+              else
+              {
+                // If a client is trying to use a newer mode, that doesn't exist, this default will
+                // be used since it should always work. This default is better than nothing and will
+                // be logged so we know to tell the user to upgrade.
+                if (Sage.DBG) System.out.println("MediaServer remux mode not supported;" +
+                    " defaulting to AUTO TRUE");
+                remuxer = new MediaServerRemuxer(fileChannel, outputFormat, true, this);
+              }
+
+              commBufWrite.clear();
+              commBufWrite.put(OK_BYTES).flip();
+            }
+
+            int numWritten = s.write(commBufWrite);
+            if (MEDIA_SERVER_DEBUG) System.out.println("MediaServer wrote out " + numWritten + " bytes");
+          }
+          else if (tempString.indexOf("REMUX_CONFIG ") == 0)
+          {
+            if (remuxer != null)
+            {
+              String config = tempString.substring(13);
+              if (config.equals("INIT"))
+              {
+                commBufWrite.clear();
+                commBufWrite.put((remuxer.isInitialized() ? "TRUE\r\n" : "FALSE\r\n").getBytes()).flip();
+              }
+              else if (config.equals("SWITCHED"))
+              {
+                commBufWrite.clear();
+                commBufWrite.put((remuxer.isSwitched() ? "TRUE\r\n" : "FALSE\r\n").getBytes()).flip();
+              }
+              else if (config.equals("FORCE_SWITCHED"))
+              {
+                remuxer.forceSwitched();
+                commBufWrite.clear();
+                commBufWrite.put(OK_BYTES).flip();
+              }
+              else if (config.equals("DISABLE_ASSIST"))
+              {
+                remuxer.disableInterAssist();
+                commBufWrite.clear();
+                commBufWrite.put(OK_BYTES).flip();
+              }
+              else if (config.equals("FORMAT"))
+              {
+                commBufWrite.clear();
+                commBufWrite.put((remuxer.isInitialized() ?
+                    remuxer.getContainerFormat().getFullPropertyString() + "\r\n" : "NULL\r\n").getBytes()).flip();
```

Comment:
Use ContainerFormat.getFullPropertyString(false). Otherwise it may include metadata which could potentially have \r\n chars in it (although in reality I don't think that could ever happen; but it's worth changing it just in case)